packages/wdio-browserstack-service/src/service.ts

```diff
@@ -526,4 +528,14 @@ export default class BrowserstackService implements Services.ServiceInstance {
 
         return (await this._browser.execute<T, []>(script))
     }
+
+    private saveWorkerData() {
+        try {
+            DataStore.saveWorkerData({
```

Comment:
Why would throw this method an exception?

Reply:
The saveWorkerData method does from file operations mkdirSync and writeFileSync which can throw the exception in any of it fails in doing so

Reply:
I would suggest to have this try/catch block in the `saveWorkerData` method then.

Reply:
Done. Moved the try catch over there